docs/feature-gates.md

```diff
@@ -188,3 +189,52 @@ Node with IP Address 10.10.10.10.
 #### Requirements for this Feature
 
 This feature is currently only supported for Nodes running Linux with IPv4 addresses.
+
+### Egress
+
+`Egress` enables a CRD API for Antrea that supports specifying which egress
+(SNAT) IP the traffic from the selected Pods to the external network should use.
+When a selected Pod accesses the external network, the egress traffic will be
+tunneled to the Node that hosts the egress IP if it's different from the Node
+that the Pod runs on and will be SNATed to the egress IP when leaving that Node.
+Usage example:
+
+```yaml
+apiVersion: crd.antrea.io/v1alpha2
+kind: Egress
+metadata:
+  name: egress-web
+spec:
+  appliedTo:
+    podSelector:
+      matchLabels:
+        role: web
+    namespaceSelector:
+      matchLabels:
+        env: prod
+  egressIP: 10.0.10.8
+```
+
+The `appliedTo` field specifies the grouping criteria of Pods to which the
+Egress applies to. Pods can be selected cluster-wide using `podSelector`. If set
+with a `namespaceSelector`, Pods from Namespaces selected by the
+namespaceSelector will be selected. The field is mandatory.
+
+The `egressIP` field specifies the egress (SNAT) IP the traffic from the
+selected Pods to the external network should use. **The IP must be assigned to
+an arbitrary interface of one Node.** It can be an IPv4 address or IPv6 address.
```

Comment:
The IP must be assigned to an arbitrary interface of one Node, and one Node only

Reply:
Do we assume we need an another user doc? There can be other details to add. For example, the IP must be reachable from all Nodes (as we tunnel to that IP).

Reply:
If this is a requirement we should list here. I'm unfamiliar with the implementation so I was assuming we were tunneling using the K8s Node IP, not the SNAT IP.

Another more detailed document (with a diagram for example) may be useful, but it's fine to tackle that after the release IMO.

Reply:
Added the requirement about reachability, agree we could have a separate doc to describe the implementation later.